src/ejabberd_config.erl

```diff
@@ -876,30 +898,104 @@ reload_cluster() ->
+                          override_local  = true, override_acls = true},
+    ConfigVersion = compute_config_version(get_local_config(),
+                                           get_host_local_config()),
+    case reload_softly_locally(ConfigDiff, State1, ConfigVersion) of
+        {error, _, Msg} ->
+            {ok, msg("Failed to apply config on node ~p: ~p", [node(), Msg])};
+        {ok, _} ->
+            RPCResult = rpc:multicall(nodes(), ?MODULE, reload_softly_remote, [ConfigFile, ConfigDiff]),
+            prepare_result(RPCResult)
+    end.
+
+
+reload_cluster_none() ->
+    CurrentNode = node(),
+    ConfigFile = get_ejabberd_config_path(),
+    State0 = parse_file(ConfigFile),
+    ConfigDiff = {CC, LC, LHC} = get_config_diff(State0),
+    ConfigVersion = compute_config_version(get_local_config(),
+                                           get_host_local_config()),
+    ?WARNING_MSG("cluster config reload from ~s scheduled", [ConfigFile]),
+    State1 = State0#state{override_global = true,
+                          override_local  = true, override_acls = true},
+    case catch apply_changes(CC, LC, LHC, State1, ConfigVersion) of
+        {ok, CurrentNode} ->
+            %% apply on other nodes
+            RPCResult = rpc:multicall(nodes(), ?MODULE, apply_changes_remote_unsafe,
+                                   [ConfigFile, ConfigDiff],
                                    30000),
-            {S1, F1} = group_nodes_results([{ok, node()} | S], F),
-            ResultText = (groups_to_string("# Reloaded:", S1)
-                          ++ groups_to_string("\n# Failed:", F1)),
-            case F1 of
-                [] -> ?WARNING_MSG("cluster config reloaded successfully", []);
-                [_ | _] ->
-                    FailedUpdateOrRPC = F ++ [Node || {error, Node, _} <- S],
-                    ?WARNING_MSG("cluster config reload failed on nodes: ~p",
-                                 [FailedUpdateOrRPC])
-            end,
-            {ok, ResultText};
+            prepare_result(RPCResult);
         Error ->
-            Reason = msg("failed to apply config on node: ~p~nreason: ~p",
-                         [CurrentNode, Error]),
-            ?WARNING_MSG("cluster config reload failed!~n"
-                         "config file: ~s~n"
-                         "reason: ~ts", [ConfigFile, Reason]),
-            exit(Reason)
+            prepare_fail_result(Error, ConfigFile)
     end.
 
+reload_softly_locally({CC, LC, LHC}, State1, ConfigVersion) ->
+    reload_softly(fun apply_changes/5, [CC, LC, LHC, State1, ConfigVersion]).
+
+reload_softly_remote(ConfigFile, ConfigDiff) ->
+    reload_softly(fun apply_changes_remote_unsafe/2, [ConfigFile, ConfigDiff]).
+
+reload_softly(ApplyChangesFun, Args) ->
+    case is_config_file_fresh() of
+        false ->
+            {error, node(), "Config file was modified before current config was laoded and `soft` safety check was chosen."};
```

Comment:
"The config file was modified before the current config was loaded and  the`soft` safety check was chosen."};